viabel/_mc_diagnostics.py

```diff
@@ -64,30 +64,30 @@ def ess(samples):
 
     rho_hat_t = np.zeros(n_draw)
     rho_hat_even = 1.0
-    rho_hat_t[0] = rho_hat_even
+    rho_hat_t =rho_hat_t.at[0].set(rho_hat_even)
```

Comment:
do you even need the equality if you are using `.set`? This applies to lines 69, 77, 78, etc. too